### PR TITLE
[HUD] Add job-status "QUEUED"

### PR DESCRIPTION
**Pull request by yangw-dev**
# Overview
https://github.com/pytorch/test-infra/issues/5822
Add Queued status

# Details
- add `Queued` value in conclusion based on job.status
- add the UI changes in both the commit view, and the main trunk view.
- replace all string such as 'pending` to enum value JobStatus 

# Notice
there are other sqls are using 'pending' as default conclusion for empty conclusion value , may create followup issues to address those.

# Next Steps
  replace the text to react-icons for rendering.

### CODE REVIEW

**Comment by huydhn** [comment]
At first glance, I think we should keep the same familiar yellow ? for pending job as the majority of them will be in pending state (screenshot).  IMO, this helps me keep the familiar context when I see them and only focus on the queueing part that stands out:

Screenshot 2024-11-04 at 21 28 24

**Comment by huydhn** [comment]
I think you do have the information about the total number of jobs in the queue.  An interesting feature here would be to use the same gray color as the ~ character for queued job as long as the their number is lower than a threshold, i.e. 50 jobs in the queue.  This signifies that everything operates normally.

On the other hand, if the number of queue jobs is higher than that threshold, meaning something is not right, we can switch the color to orange, and it would surely catch our attention.

This can be done in a follow-up PR though I think

File: torchci/clickhouse_queries/hud_query/query.sql [hunk 1]
```diff
@@ -47,11 +48,15 @@ SELECT
     sha,
     CONCAT(workflow_name, ' / ', job_name) as name,
     id,
-    if(
-      conclusion = '',
-      'pending',
-      conclusion
-     ) as conclusion,
+    multiIf(
```
Comment (huydhn) [inline torchci/clickhouse_queries/hud_query/query.sql:51]: TIL!

**Comment by huydhn** [comment]
Nit: Could we get rid of the show log part when the job is pending or in queue as there is nothing to show there yet except and S3 error that the key doesn't exist?

Screenshot 2024-11-04 at 21 44 20

**Comment by yangw-dev** [comment]
> character

Hi Huy, I will replace all of those text with react-icon in next PR, so this is temporary, we can def discuss the color change in next PR!

**Comment by yangw-dev** [comment]
> Nit: Could we get rid of the show log part when the job is pending or in queue as there is nothing to show there yet except and S3 error that the key doesn't exist?
> 
> Screenshot 2024-11-04 at 21 44 20

I agree the log is not very useful here, @clee2000 , thoughts?

File: torchci/lib/JobClassifierUtil.ts [hunk 2]
```diff
@@ -215,8 +215,10 @@ export function getGroupConclusionChar(conclusion?: GroupedJobStatus): string {
       return "O";
     case GroupedJobStatus.Failure:
       return "X";
+    case GroupedJobStatus.Queued:
+      return "Q";
     case GroupedJobStatus.Pending:
-      return "?";
+      return "P";
```
Comment (huydhn) [inline torchci/lib/JobClassifierUtil.ts:221]: I would prefer to keep ? here (until you change it properly to use react icons), people are familiar with seeing that for pending jobs

Comment (yangw-dev) [reply torchci/lib/JobClassifierUtil.ts:221]: Hi Huy, I'm thinking replace those with react-icons in next PR, but def, i can change it back to this pr

Comment (huydhn) [reply torchci/lib/JobClassifierUtil.ts:221]: Yeah, I think this change goes beyond scope of this PR which is to add the queued status.  So, it's better to save it on the next PR as you plan to replace those anyway.  We can ask more people to look at the next PR to gather their opinions

Comment (yangw-dev) [reply torchci/lib/JobClassifierUtil.ts:221]: sounds good! Will add both Q and this change to what you suggest

**Comment by yangw-dev** [comment]
> I think you do have the information about the total number of jobs in the queue. An interesting feature here would be to use the same gray color as the ~ character for queued job as long as the their number is lower than a threshold, i.e. 50 jobs in the queue. This signifies that everything operates normally.
> 
> On the other hand, if the number of queue jobs is higher than that threshold, meaning something is not right, we can switch the color to orange, and it would surely catch our attention.
> 
> This can be done in a follow-up PR though I think

I changed the queued signal to grey

**Review by huydhn — APPROVED** [review]

File: torchci/lib/JobClassifierUtil.ts [hunk 3]
```diff
@@ -265,8 +286,10 @@ export function getConclusionChar(
       return "T";
     case JobStatus.Skipped:
       return "S";
+    case JobStatus.Queued:
+      return "Q";
     case JobStatus.Pending:
-      return "?";
+      return "P";
```
Comment (huydhn) [inline torchci/lib/JobClassifierUtil.ts:292]: Nit: I think we want to keep ? here too to be consistent with the one above

**Review by clee2000 — APPROVED** [review]
Personally I prefer yellow over gray for queued, but given you're going to change the icons in a different PR I don't think it matters much

**Pull request merged**